File: Backjoon4948/Backjoon4948.py
def ber(n): 
    nm = list(range(n + 1, 2 * n + 1))
    num = n 
    count = 0

    if(nm[0] == 1):
        nm[0] = 0
            
    for i in range(2, n):
        for ii in range(0, num):
            if(nm[ii] != 0 and nm[ii] % i == 0):
                nm[ii] = 0

    for i in range(0, num):
        if(nm[i] != 0):
            nm[count] = nm[i]
            count += 1

    for i in range(0, count):
        if(nm[i] != 0):
            for ii in range(i + 1, count):
                if(nm[ii] != 0 and nm[ii] % nm[i] == 0):
                    nm[ii] = 0

    num = 0
    for i in range(0, count):
        if(nm[i] != 0):
            nm[num] = nm[i]
            num += 1

    print(num)

n = list()

n.append(1)
n.append(10)
n.append(13)
n.append(100)
n.append(1000)
n.append(10000)
n.append(100000)

for i in n:
    ber(i)

File: Baekjoon_step/Backjoon4948/Backjoon4948.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h> 

void ber(int n){
    int num = n;
    int*nm = (int*)malloc(sizeof(int) * (num));
    int count = 0;
    
    for(int i = 1; i < num; i++)
        nm[i] = i + n;

    if(nm[0] == 1) 
        nm[0] = 0;

    for(int i = 2; i < n; i++)
        for(int ii = 0; ii < num; ii++)
            if(nm[ii] != 0 && nm[ii] % i == 0) 
                nm[ii] = 0;

    for(int i = 0; i < num; i++)
        if(nm[i] != 0)
            nm[count++] = nm[i];

    for(int i = 0; i < count; i++)
        if(nm[i] != 0)
            for(int ii = i + 1; ii < count; ii++)
                if(nm[ii] != 0 && nm[ii] % nm[i] == 0)
                    nm[ii] = 0;

    num = 0;
    for(int i = 0; i < count; i++)
        if(nm[i] != 0)
            nm[num++] = nm[i];

    printf("%d\n", num);
}

int main(){
    int n [100] = {1, 10, 13, 100, 1000, 10000, 100000};
    int count = 7;
    for(int i = 0; i < count; i++)
        ber(n[i]);
}